src/main/java/net/minecraftforge/fluids/BlockFluidBase.java

```diff
@@ -453,6 +453,13 @@ public boolean shouldSideBeRendered(@Nonnull IBlockState state, @Nonnull IBlockA
         return super.shouldSideBeRendered(state, world, pos, side);
     }
 
+    public boolean isCoveredWithFluid(IBlockAccess world, BlockPos pos)
+    {
+        IBlockState here = world.getBlockState(pos);
+        IBlockState up = world.getBlockState(pos.down(densityDir));
+        return here.getBlock() == this && (up.getMaterial().isLiquid() || up.getBlock() instanceof IFluidBlock);
+    }
```

Comment:
I think it should match vanilla. If we get away from vanilla there are all kinds of small improvements we might want.